server/src/main/java/com/ca/lsp/core/cobol/semantics/PredefinedVariableContext.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (c) 2020 Broadcom.
+ * The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *    Broadcom, Inc. - initial API and implementation
+ *
+ */
+
+package com.ca.lsp.core.cobol.semantics;
+
+import com.ca.lsp.core.cobol.model.Variable;
+import com.google.common.collect.HashMultimap;
+import com.google.common.collect.Multimap;
+import lombok.EqualsAndHashCode;
+import lombok.NoArgsConstructor;
+import org.eclipse.lsp4j.Location;
+
+import java.util.List;
+
+/**
+ * This class represents a structure to store variables of a COBOL program and build a variable
+ * hierarchy.
+ */
+@EqualsAndHashCode
+@NoArgsConstructor
+public class PredefinedVariableContext implements SubContext<Variable> {
+  private static final Location MOCK_LOCATION = new Location();
+  private static final Multimap<String, Location> VARIABLE_DEFINITIONS = HashMultimap.create();
+
+  private final Multimap<String, Location> variableUsages = HashMultimap.create();
+
+  static {
+    // EXEC interface block fields
+    VARIABLE_DEFINITIONS.put("EIBAID", MOCK_LOCATION);
```

Comment:
What's about constructing a list and then convert it into a map?
Like you did in `server/src/test/java/com/ca/lsp/cobol/usecases/engine/UseCasePreprocessorListener.java`.

Reply:
I wanted to avoid unnecessary objects in the production code